topics-and-advance-readings/psdad.md

```diff
@@ -0,0 +1,213 @@
+>
+> **temp**: The "temp" field specifies the temperature measured at the given probe, rounded off to the nearest integer, in degrees Celsius.
+>
+> ...
+
+We want that spec text to be connected to the bits on the wire, in a
+way that is secure, scalable, and maintainable. Operationally, we want
+the people writing any software which might be working with the data
+in those fields to be looking at the same spec text, with minimal
+chance for error or disinformation.
+
+The PSDAD approach is to send that spec text on the wire wrapped
+around the data fields.  For example, Alice might transmit:
+
+```text
+The temperature at probe number 6 is 34°C.
+```
+
+To a human who knows English, this is fairly meaningful.  To the
+software at either end, this looks pretty much the same as the JSON
+expression given above, `{"probe":6,"temp": 34}`.  It's a few bytes
+longer, but it's still "stuff 6 stuff 34 stuff".
+
+The PSDAD trick is to make the delimiters be the documentation by
+writing the documentation as template assertions, then filling in the
+templates with the field data values.
+
+The template in use here is `The temperature at probe number [] is
+[]°C.` Given that template, software at either end can easily
+serialize and deserialize probe temperature data.
+
+## Applicability
+
+The primary use case here is to have decentralized extensibility be
+robust against disinformation attacks, but this technique also has
+potential to address schema and documentation management issues in
+conventional deployment environments. In particular, PSDAD allows
+developers to publish and/or consume machine-readable data with semantic
+clarity and minimal procedural work or need for prior agreement.
+
+If you want to publish some data, just think about how you would
+express each item in that data clearly, pedantically, as a statement
+in your preferred natural language.  If you want to consume some data,
+look at some examples and deduce the template.  If it's not obvious,
+there's probably an issue.
+
+Here's another example.  Alice wants to help build a web of trust, so
+she publish this text at https://alice.example/endorsements :
+
+```text
+I have met and know the person "Susan X. Lastname".
+Susan X. Lastname controls and is the sole user of the website https://susan.example.
+Susan X. Lastname has the ssh key with public part ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDQnL/9t1t+w8U9bBOn3OeM568BkiejgK37ITAgaRHyGw0+vYCqinMKeswzv0YFar8n9M+Rwi78Evk72RuOlQldcw5cmVfrgwey3U7k/0cJE5ecnO2CEBU4zqfBiKlGnlRsjQ1UcsY2e396ScBGrzm02UYBmtRa09fA+vbSxN4O6nHVDdR1cscbOa3TXhfPpp009LdDzUIt5AgiwXPdBYOBBrvRnzULhGMWiTy8HEXpueRpBOa90Q7yEhL/zZluZQo8tMRUYkk6EwTvJUmz3TjrYKI5DO5p5q1mPR1/wMjVRUEwYG7HET6PgQuJj4aQiIrPQZbVftd+AVOjJyvenkW9 susan@susan.example.
+
+I have met and know the person Tommy Lastname.
+Tommy Lastname controls and is the sole user of the website https://tommy.example.
+```
+
+These strings can be serialized and/or deserialized easily, given these templates:
```

Comment:
Digital Bazaar specifically is interested in "templated semantics" in the "Verifiable Credentials" sense. We are attempting to address this problem by:

1. Making it easier to publish and version vocabularies and JSON-LD contexts.
2. Creating visual templates, via "Web Components", so that people can fill these templates out.

In our mind, the problem has more to do w/ tooling than a new language to express these concepts. However, we are very interested in exploring multiple paths to solving the problem.

The most interesting things to me are: 1) the "natural language as the term" component to these proposals as well as the 2) template mechanism. Feels like there is something there.